packages/shared/src/utils/route/route.tsx

```diff
@@ -0,0 +1,61 @@
+// Checks if pathname matches route. (Works even with query string /?)
+
+// TODO: Add test cases for this
+import React from 'react';
+import { Redirect } from 'react-router-dom';
+import { deepCopy } from '../object';
+import { routes } from '../routes';
+
+export type TRoute = {
+    component?: React.ElementType | null | ((routes?: TRoute[]) => JSX.Element) | typeof Redirect;
+    default?: boolean;
+    exact?: boolean;
+    getTitle?: () => string;
+    icon_component?: string;
+    id?: string;
+    is_authenticated?: boolean;
+    is_invisible?: boolean;
+    path?: string;
+    to?: string;
+    icon?: string;
+    is_disabled?: boolean;
+    subroutes?: TRoute[];
+    routes?: TRoute[];
+};
+
+type TGetSelectedRoute = {
+    routes: TRoute[];
+    pathname: string;
+};
+
+// @ts-expect-error as this is a utility function with dynamic types
+export const matchRoute = <T,>(route: T, pathname: string) => new RegExp(`^${route?.path}(/.*)?$`).test(pathname);
```

Comment:
```suggestion
export const matchRoute = (route: TRoute, pathname: string) => new RegExp(`^${route?.path}(/.*)?$`).test(pathname);
```

I think most of the args are known so u could probably type-safe this

Reply:
Unfortunately it leads to failing build for trader package. :(